src/disk_io_thread.cpp

```diff
@@ -658,7 +658,7 @@ namespace libtorrent
 		if (!failed)
 		{
 			TORRENT_PIECE_ASSERT(!error, pe);
-			std::uint32_t write_time = total_microseconds(clock_type::now() - start_time);
+			int const write_time = int(total_microseconds(clock_type::now() - start_time));
```

Comment:
I made all these calls the same (cast to `int`) since some of them already were `int`, but do you think it's better to take the `std::int64_t` and cast just in `add_sample`?